docs: 0.13 update for docker compose migration guide (#4235)
garden-io/garden: docs/guides/migrating-from-docker-compose.md

  - name: local-kubernetes
```

This is a `Project` level file. We call it `compose2garden` in our example, but you can use your own name. We configure
a single environment and specify the hostname where we can visit the running application. Finally, we
configure `local-kubernetes` (a Kubernetes cluster running in Docker Desktop) as our provider.

### The `backend/backend.garden.yml` file

For our `backend` application, we've added another Garden configuration file:

```yaml
kind: Build
apiVersion: garden.io/v1
name: backend
description: The backend server image
type: container

---
kind: Deploy
apiVersion: garden.io/v1
name: backend
description: The backend server container
type: container
build: backend
dependencies:
  - deploy.mongo
spec:
  sync:
    paths:
      - source: ./
        target: /usr/src/app
        mode: "one-way-replica"
  ports:
    - name: http
      containerPort: 3000
  healthCheck:
    httpGet:
      path: /api
      port: http
  ingresses:
    - path: /
      port: http
      hostname: backend.${var.base-hostname}
```

> It defines 2 actions: a trivial `Build` action, and a `Deploy` action.

Reviewer: ```suggestion
A `Build` action and a `Deploy` action are defined.
```

> Let's consider `Deploy` action in a bit more details.

Reviewer: ```suggestion
Make note of the `Deploy` action and it's configuration.
```

> Under `sync` we set up syncing from the root folder to our built `app` folder, so we can see code changes live when in
> developer mode.

Reviewer: ```suggestion
Under `sync` we set up syncing from the action root to the `app` folder on the container, so we can synchronize code changes live when in
`sync` mode.
```

> Under `sync` we set up syncing from the root folder to our built `app` folder, so we can see code changes live when in
> developer mode.

Reviewer: ```suggestion
Under `sync` we set up syncing from the action root to the `app` folder on the container, so we can synchronize code changes live when in
`sync` mode.
```

Author: This has been committed.

Under `ports` we specify the same port as in our Docker Compose file (`3000`).

We set up a health check for the `/api` route, as that is what this module serves, and an ingress on a subdomain. In our
case, this will let us access our `backend` application on `compose2garden.local.demo.garden`.

> Finally, we specify the dependency on the `mongo` module, which we will define in a bit.

Reviewer: Modules have dropped out of style

### The `frontend/frontend.garden.yml` file

> For the `frontend` we create another Garden configuration file:

Reviewer: ```suggestion
For the `frontend` application we create separate Garden configuration file:
```

```yaml
kind: Build
apiVersion: garden.io/v1
name: frontend
description: The frontend server and UI components image
type: container
exclude:
  - node_modules/**/*

---
kind: Deploy
apiVersion: garden.io/v1
name: frontend
description: The frontend server and UI components container
type: container
build: frontend
dependencies:
  - deploy.backend
spec:
  env:
    DANGEROUSLY_DISABLE_HOST_CHECK: true
  sync:
    paths:
      - source: ./src
        target: /usr/src/app/src
        mode: "one-way-replica"
  ports:
    - name: http
      containerPort: 3000
  healthCheck:
    httpGet:
      path: /
      port: http
  ingresses:
    - path: /
      port: http
```

This is similar to the `backend` application, but we specify the `backend` deployment as a dependency, which makes
the database (`mongo`) an indirect dependency.

### The `mongo/mongo.garden.yml` file

> Here we've created a `mongo` folder, as it did not exist as an explicit module in our original Docker Compose project.

Reviewer: Module

Author: Changed this in https://github.com/garden-io/garden/pull/4235/commits/f1e1c0cbc989c1e828031c7fe65db59ae0c947ba

  ingresses:
    - path: /
      port: http
```

This is similar to the backend module, but we specify the backend as a dependency, which makes the database (`mongo`) an indirect dependency.
This is similar to the `backend` application, but we specify the `backend` deployment as a dependency, which makes
the database (`mongo`) an indirect dependency.

### The `mongo/mongo.garden.yml` file

Here we've created a `mongo` folder, as it did not exist as an explicit module in our original Docker Compose project. The folder contains only the Garden configuration file, which contains:
Here we've created a `mongo` folder, as it did not exist as an explicit module in our original Docker Compose project.
The folder contains only the Garden configuration file:

```yaml
kind: Module
description: Mongodb for storing todo items
kind: Deploy
apiVersion: garden.io/v1
description: MongoDB for storing todo items
type: container
name: mongo
image: mongo:4.2.0
services:
  - name: mongo
    volumes:
      - name: data
        containerPath: /data/db
    ports:
      - name: db
        containerPort: 27017
```

This specifies the same volume and port that we previously specified in Docker Compose.

> ## Running a development version of the project on Garden

Reviewer: I think you should add a note about `garden dev` and `sync` mode somewhere as a replacement for this block.

Author: Yes, we can put a link to the code sync guide here, and not describe too many details.

Author: Restored it in https://github.com/garden-io/garden/pull/4235/commits/f9ff0fdde7af965b451bc0f854c4873e33b8bc1d

> This was a simple example, but it should give you what you need to migrate larger projects too. If you have feedback on
> how we could make migrating from Docker Compose easier, please send us feedback
> via [GitHub issues](https://github.com/garden-io/garden/issues), or reach out
> on [our Discord community](https://discord.gg/FrmhuUjFs6).

Reviewer: ```suggestion
This is a basic example but it should give you what you need to migrate larger projects too. If you have feedback on
how we could make migrating from Docker Compose easier, please send it our way
via [GitHub issues](https://github.com/garden-io/garden/issues) or reach out
on [our Discord community](https://discord.gg/FrmhuUjFs6).
```

I don't really like the "should give you what you need to" part but couldn't think of anything better.
